plugins/sbt/src/index.ts

```diff
@@ -0,0 +1,198 @@
+      return cleanOutput;
+    }
+
+    /** Read version from sbt */
+    async function sbtGetVersion() {
+      // in multi-module projects, we want to get only ThisBuild/version
+      await sbtClient("set version/aggregate := false");
+      const output = await sbtClient("print version");
+      const version = output.split("\n").shift()?.trim();
+      if (!version) {
+        throw new Error(`Failed to read version from sbt: ${output}`);
+      }
+
+      auto.logger.log.info(`Got version from sbt: ${version}`);
+      return version;
+    }
+
+    /** Set version in sbt to the given value */
+    async function sbtSetVersion(version: string) {
+      auto.logger.log.info(`Set version in sbt to "${version}"`);
+      return sbtClient(`set every version := \\"${version}\\"`);
+    }
+
+    /** Run sbt publish */
+    async function sbtPublish() {
+      auto.logger.log.info("Run sbt publish");
+      const publishLog = await sbtClient("publish");
+      auto.logger.log.info("Output:\n" + publishLog);
+      return publishLog;
+    }
+
+    /** Construct canary version using Auto-provided suffix */
+    async function getCanaryVersion(canaryIdentifier: string) {
+      const lastTag = await getTag();
+      const lastVersion = lastTag.replace(/^v/, "");
+      return `${lastVersion}${canaryIdentifier}-SNAPSHOT`;
+    }
+
+    auto.hooks.validateConfig.tapPromise(this.name, async (name, options) => {
+      // If it's a string thats valid config
+      if (name === this.name && typeof options !== "string") {
+        return validatePluginConfiguration(this.name, pluginOptions, options);
+      }
+    });
+
+    // exact copy-paste from the git-tag plugin
+    auto.hooks.getPreviousVersion.tapPromise(this.name, async () => {
+      if (!auto.git) {
+        throw new Error(
+          "Can't calculate previous version without Git initialized!",
+        );
+      }
+
+      return getTag();
+    });
+
+    // exact copy-paste from the git-tag plugin
+    auto.hooks.version.tapPromise(
+      this.name,
+      async ({ bump, dryRun, quiet }) => {
+        if (!auto.git) {
+          return;
+        }
+
+        const lastTag = await getTag();
+        const newTag = inc(lastTag, bump as ReleaseType);
+
+        if (dryRun && newTag) {
+          if (quiet) {
+            console.log(newTag);
+          } else {
+            auto.logger.log.info(`Would have published: ${newTag}`);
+          }
+
+          return;
+        }
+
+        if (!newTag) {
+          auto.logger.log.info("No release found, doing nothing");
+          return;
+        }
+
+        const prefixedTag = auto.prefixRelease(newTag);
+
+        auto.logger.log.info(`Tagging new tag: ${lastTag} => ${prefixedTag}`);
+        await execPromise("git", [
+          "tag",
+          prefixedTag,
+          "-m",
+          `"Update version to ${prefixedTag}"`,
+        ]);
+
+        if (this.options.manageVersion) {
+          await sbtSetVersion(newTag);
```

Comment:
I think you'll want to do this before tagging so that the version is included in the tag

Reply:
unless `sbt` publish is doing the committing here

Reply:
It's a bit different here. Doesn't persist `version` between the builds. It's possible to set it statically, but it's not the idiomatic way. So there are plugins for versioning, like the [sbt-dynver](https://github.com/dwijnand/sbt-dynver) I recommended in the readme, that calculate version based on `git describe`. 

After we create a git tag and "reload" sbt, it will calculate the `version` setting based on the tag (it will be the same, but without `v` prefix).

If we don't create a tag, like in the canary case, `sbt-dynver` will set it to `git describe`. But if we don't use `sbt-dynver`, I provide this `manageVersion` setting as a fallback and it will explicitly set the version to the Auto-style canary just before publishing.

Reply:
> Doesn't persist `version` between the builds.

Also, because of this, there is no committing involved at any point.

Reply:
oh okay, good to know!